robotidy/transformers/GenerateDocumentation.py

```diff
@@ -0,0 +1,187 @@
+
+
+class KeywordData:
+    def __init__(self, name, arguments, returns):
+        self.name = name
+        self.arguments = arguments
+        self.returns = returns
+
+
+class FormattingData:
+    def __init__(self, cont_indent, separator):
+        self.cont_indent = cont_indent
+        self.separator = separator
+
+
+class ArgumentsAndsReturnsVisitor(ModelVisitor):
+    def __init__(self):
+        self.arguments = []
+        self.returns = []
+        self.doc_exists = False
+
+    def visit_Keyword(self, node):  # noqa
+        self.arguments = []
+        self.returns = []
+        # embedded variables
+        for variable in node.header.data_tokens[0].tokenize_variables():
+            if variable.type == Token.VARIABLE:
+                self.arguments.append(Argument(variable.value))
+        self.doc_exists = False
+        self.generic_visit(node)
+
+    def visit_Documentation(self, node):  # noqa
+        self.doc_exists = True
+
+    def visit_Arguments(self, node):  # noqa
+        if node.errors:
+            return
+        self.arguments = [Argument(arg) for arg in node.values]
+
+    def visit_ReturnStatement(self, node):  # noqa
+        if node.errors:
+            return
+        self.returns = list(node.values)
+
+    visit_Return = visit_ReturnStatement
+
+
+class GenerateDocumentation(Transformer):
+    """
+    Generate keyword documentation with the documentation template.
+
+    By default, GenerateDocumentation uses Google documentation template.
+    Following keyword:
+
+    ```robotframework
+    *** Keywords ***
+    Keyword
+        [Arguments]    ${arg}
+        ${var}   ${var2}    Step
+        RETURN    ${var}    ${var2}
+    ```
+
+    will produce following documentation:
+
+    ```robotframework
+    *** Keywords ***
+    Keyword
+        [Documentation]
+        ...
+        ...    Arguments:
+        ...        ${arg}:
+        ...
+        ...    Returns:
+        ...        ${var}
+        ...        ${var2}
+        [Arguments]    ${arg}
+        ${var}   ${var2}    Step
+        RETURN    ${var}    ${var2}
+    ```
+
+    It is possible to create own template and insert dynamic text like keyword name, argument default values
+    or static text (like ``[Documentation]    Documentation stub``). See our docs for more details.
+
+    Generated documentation will be affected by ``NormalizeSeparators`` transformer that's why it is best to
+    skip formatting documentation by this transformer:
+
+    ```
+    > robotidy --configure GenerateDocumentation:enabled=True --configure NormalizeSeparators:skip_documentation=True src
+    ```
+    """
+
+    ENABLED = False
+
+    WHITESPACE_PATTERN = re.compile(r"(\s{2,}|\t)", re.UNICODE)
+
+    def __init__(self, overwrite: bool = False, doc_template: str = "google", template_directory: Optional[str] = None):
+        self.overwrite = overwrite
+        self.doc_template = Template(self.get_template(doc_template, template_directory))
```

Comment:
I think you need to add an error checking when loading template - if it is correct or not.

Reply:
Oh, totally forgot about possible errors when rendering jinja template. Thanks, I will add it